types/proposal.go

```diff
@@ -54,9 +55,15 @@ func (p *Proposal) ValidateBasic() error {
 	if p.Round < 0 {
 		return errors.New("negative Round")
 	}
+	if p.Round > math.MaxInt32 {
```

Comment:
these checks do not make sense. p.Round won't ever be greater than MaxInt32.

What would be good is to add a check: when we increment Round/POLRound and Round == math.MaxInt32, we panic.